From f3b927017ecced24a04871c148cd5b3194d3ca65 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Tue, 10 Mar 2026 11:19:03 +0100
Subject: [PATCH 10/12] bindings: Add ibm,spyre binding

Add binding for IBM's AI accelerator "Spyre" that has an I2C interface
to gather temperature readings from a managment controller.

Upstream-Status: Pending
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 .../devicetree/bindings/hwmon/ibm,spyre.yaml  | 40 +++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/hwmon/ibm,spyre.yaml

diff --git a/Documentation/devicetree/bindings/hwmon/ibm,spyre.yaml b/Documentation/devicetree/bindings/hwmon/ibm,spyre.yaml
new file mode 100644
index 000000000000..a117d610a305
--- /dev/null
+++ b/Documentation/devicetree/bindings/hwmon/ibm,spyre.yaml
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/hwmon/ibm,spyre.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: IBM Spyre AI accelerator temperature sensor
+
+maintainers:
+  - Krzysztof Kozlowski <krzk@kernel.org>
+
+properties:
+  compatible:
+    enum:
+      - ibm,spyre-i2c
+
+  reg:
+    maxItems: 1
+
+  "#thermal-sensor-cells":
+    const: 1
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        sensor@48 {
+            compatible = "ibm,spyre-i2c";
+            reg = <0x24>;
+            #thermal-sensor-cells = <1>;
+        };
+    };
-- 
2.43.0

